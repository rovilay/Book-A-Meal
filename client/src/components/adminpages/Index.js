import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import navData from '../../helpers/navData';
import { setNav } from '../../actions/navLinks';


class MenuPage extends Component {
  componentDidMount() {
    this.props.dispatch(setNav(navData.adminNavDefault));
  }
  render() {
    return (
      <div>
        admin Page!
      </div>
    );
  }
}

MenuPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(MenuPage);
