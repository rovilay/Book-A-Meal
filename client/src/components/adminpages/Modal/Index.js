import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

import MenuDetailsTable from '../dashboard/MenuTable/MenuDetails';
import EditMenuTable from '../dashboard/MenuTable/EditMenu';
import OrderDetailsTable from '../orderPage/orderDetails';

Modal.setAppElement('#root');
class ModalComp extends Component {
  componentDidMount() {
    this.props.emptyEditMenu();
  }
  render() {
    const { hideModal, modal } = this.props;
    const {
      isOpen,
      isInfo,
      isEdit,
      isOrderInfo,
      content,
      contentLabel,
    } = modal;

    return (
      <Modal
        isOpen={isOpen}
        contentLabel= {contentLabel}
        overlayClassName="overlay"
        className="modal"
      >
        <div className="closeBtn">
          <button
            onClick={hideModal}
            className="btn-col btn-1"
          >
            close
          </button>
        </div>
        {
          (isInfo && content)
          &&
          (
            <MenuDetailsTable
              title={contentLabel}
              content={content}
              {...this.props}
            />
          )
        }

        {
          (isEdit)
          &&
          (
            <EditMenuTable
              title={contentLabel}
              content={content}
              isEdit={isEdit}
              {...this.props}
            />
          )
        }

        {
          (isOrderInfo)
          &&
          (
            <OrderDetailsTable
              title={contentLabel}
              content={content}
              isEdit={isEdit}
              {...this.props}
            />
          )
        }
      </Modal>
    );
  }
}

ModalComp.defaultProps = {
  emptyEditMenu: undefined
};

ModalComp.propTypes = {
  modal: PropTypes.object.isRequired,
  setModal: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
  emptyEditMenu: PropTypes.func,
};

export default ModalComp;
