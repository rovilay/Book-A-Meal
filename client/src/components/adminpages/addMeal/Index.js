import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddMeal extends Component {
  render() {
    return (
      <div>
        addmeal page
      </div>
    );
  }
}

export default AddMeal;
