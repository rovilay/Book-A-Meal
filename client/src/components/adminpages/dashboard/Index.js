/*
  eslint class-methods-use-this:0,
  eslint no-alert: 0,
*/
/* eslint no-restricted-globals: 0 */
import arraySort from 'array-sort';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  Accordion,
} from 'react-accessible-accordion';

import navData from '../../../helpers/navData';
import filterify from '../../../helpers/filterify';
import { setNav } from '../../../actions/navLinksActions';
import {
  getTodayMenu, getAllMenus, postMenu,
  updateMenu, addMealToNewMenu, removeMealFromNewMenu,
  emptyNewMenu, emptyEditMenu, addMealInEditMenu,
  deleteMealInEditMenu, setMenuForEdit, getMenuMeals,
  deleteMealInMenu, deleteMenuMeal
} from '../../../actions/menuActions';
import { getMeals } from '../../../actions/mealActions';
import {
  setModal, deleteMealInEditModal,
  addMealInEditMenuModal
} from '../../../actions/modalActions';
import MenuTable from './MenuTable/MenuTable';
import SetMenuCard from './MenuCard/SetMenu';
import Filter from '../../common/Filter';
import setFilter from '../../../actions/filterActions';
import ModalComp from '../Modal/Index';

class AdminDashboard extends Component {
  constructor(props) {
    super(props);

    this.setNewMenuMeal = this.setNewMenuMeal.bind(this);
    this.submitNewMenu = this.submitNewMenu.bind(this);
    this.showMenuDetails = this.showMenuDetails.bind(this);
    this.hideModal = this.hideModal.bind(this);
    this.editMenu = this.editMenu.bind(this);
    this.deleteRow = this.deleteRow.bind(this);
    this.onSubmitUpdate = this.onSubmitUpdate.bind(this);
    this.unCheckAll = this.unCheckAll.bind(this);
    this.getMenus = this.getMenus.bind(this);
    this.handlePaginationClick = this.handlePaginationClick.bind(this);
  }

  componentDidMount() {
    this.props.setNav(navData.adminNavDefault);
    this.props.getMeals({});
    this.props.getAllMenus({});
    this.props.setFilter({ filter: 'all' });
    this.hideModal();
  }

  onSubmitUpdate(menuDate, meals) {
    this.props.updateMenu({ menuDate, meals });
  }

  setNewMenuMeal(mealId) {
    let checkbox;
    if (this.props.modal.isEdit) {
      checkbox = document.getElementById(`${mealId}-edit`);
    } else {
      checkbox = document.getElementById(mealId);
    }
    if (checkbox.checked === true) {
      this.props.addMealToNewMenu(mealId);
    } else if (checkbox.checked === false) {
      this.props.removeMealFromNewMenu(mealId);
    }
  }

  /**
   * gets menus
   * @param {number} limit pagination limit
   * @param {number} offset pagination offset
   * @param {string} menuUrl menu meals url to get meals for a menu
   */
  getMenus({ limit, offset, menuUrl }) {
    this.props.getAllMenus({
      limit,
      offset,
      menuUrl
    });
  }

  hideModal() {
    this.props.setModal({
      isOpen: false,
      isEdit: false,
      isInfo: false,
      close: true,
      contentLabel: '',
      content: {}
    });
  }

  showMenuDetails(menuDetail) {
    this.props.setModal({
      isOpen: true,
      isInfo: true,
      isEdit: false,
      close: false,
      contentLabel: 'Menu details',
      content: { menuDetails: [...menuDetail] }
    });
  }

  editMenu({ menuId, postOn, meals }) {
    getMeals({});
    this.props.setModal({
      isOpen: true,
      isInfo: false,
      isEdit: true,
      close: false,
      contentLabel: 'Add Meals',
      content: { menuId, postOn, meals }
    });
    this.props.setMenuForEdit(meals);
    this.props.emptyNewMenu();
  }

  deleteRow(id) {
    this.props.deleteMealInEditModal(id);
    this.props.deleteMealInEditMenu(id);
  }

  unCheckAll(mealIdArr) {
    mealIdArr.forEach((mealId) => {
      const checkbox = document.getElementById(mealId);
      checkbox.checked = false;
    });
  }

  submitNewMenu() {
    const meals = [...this.props.newMenuMeals];
    const date = document.getElementById('postOn').value;
    const postOn = date.split('/').reverse().join('-');
    this.props.postMenu({ postOn, meals });
  }


  /**
   * handles pagination changes
   *
   * @param {object} data data object from pagination component
   */
  handlePaginationClick(data) {
    // const nextPage = data.selected + 1;
    const { limit } = this.props.pagination;
    const offset = (data.selected) * limit;
    this.getMenus({ limit, offset });
  }

  render() {
    const { user } = this.props;
    const { firstName, lastName } = user;
    return (
      <div>
        <div className="welcome">
          <p className="merienda">
            welcome, {firstName} {lastName}
          </p>
        </div>
        <section className="setmenu">
          <Accordion>
            <div className="setmenu-container">
              <SetMenuCard
                setNewMenuMeal={this.setNewMenuMeal}
                submitNewMenu={this.submitNewMenu}
                {...this.props}
              />
            </div>
          </Accordion>
        </section>
        {/* <MenuAccordion /> */}
        <section className="adminpage">
          <div className="menu-title">Menus List</div>
          <div className="container">
            <Filter
              {...this.props}
              tableContent="Menus_List"
            />
            {
              (this.props.menus.length === 0)
                ?
                  <p className="empty not-found">No menu found!</p>
                :
                  <Accordion>
                    <MenuTable
                      showMenuDetails={this.showMenuDetails}
                      editMenu={this.editMenu}
                      {...this.props}
                    />
                  </Accordion>
            }
          </div>
        </section>
        <ModalComp
          hideModal={this.hideModal}
          deleteRow={this.deleteRow}
          submitUpdate={this.onSubmitUpdate}
          setNewMenuMeal={this.setNewMenuMeal}
          {...this.props}
        />
      </div>
    );
  }
}

AdminDashboard.propTypes = {
  setNav: PropTypes.func.isRequired,
  getMeals: PropTypes.func.isRequired,
  getAllMenus: PropTypes.func.isRequired,
  meals: PropTypes.array.isRequired,
  addMealToNewMenu: PropTypes.func.isRequired,
  removeMealFromNewMenu: PropTypes.func.isRequired,
  postMenu: PropTypes.func.isRequired,
  newMenuMeals: PropTypes.array.isRequired,
  setModal: PropTypes.func.isRequired,
  deleteMealInEditModal: PropTypes.func.isRequired,
  updateMenu: PropTypes.func.isRequired,
  setMenuForEdit: PropTypes.func.isRequired,
  emptyNewMenu: PropTypes.func.isRequired,
  modal: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired,
  menus: PropTypes.array.isRequired,
  setFilter: PropTypes.func.isRequired,
  deleteMealInEditMenu: PropTypes.func.isRequired,
  pagination: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  meals: arraySort(state.meal.meals, 'title'),
  newMenuMeals: state.menu.newMenu,
  menus: filterify(state.menu.allMenus, state.filter),
  menuMeals: state.menu.menuMeals,
  pagination: state.menu.pagination,
  menuMealsPagination: state.menu.menuMealsPagination,
  modal: state.modal,
  editMenuMeals: state.menu.editMenu,
  mealPagination: state.meal.pagination,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setNav,
    setModal,
    getMeals,
    emptyNewMenu,
    getAllMenus,
    getTodayMenu,
    updateMenu,
    postMenu,
    emptyEditMenu,
    addMealToNewMenu,
    removeMealFromNewMenu,
    setMenuForEdit,
    deleteMealInEditMenu,
    deleteMealInMenu,
    deleteMealInEditModal,
    addMealInEditMenu,
    addMealInEditMenuModal,
    getMenuMeals,
    setFilter,
    deleteMenuMeal
  },
  dispatch
);

export default connect(mapStateToProps, mapDispatchToProps)(AdminDashboard);
