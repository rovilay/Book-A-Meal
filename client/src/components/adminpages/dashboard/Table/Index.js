import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import tableHead from '../../../../helpers/tableHead';
import TableHead from '../../../common/Table/TableHead';
import MenuTableRow from './MenuTableRow';

class MenuTable extends Component {
  render() {
    const { menus } = this.props;
    return (
      <div className="menu_table">
        <table>
          <TableHead tableHead={tableHead.menuTableHead} />
          <tbody>
            {
              menus.map((menu, i) => {
                const {
                  id: menuId,
                  postOn,
                  User,
                } = menu;
                const item = {
                  sn: ++i,
                  menuId,
                  postOn: moment(postOn).format('LL'),
                  createdBy: `${User.firstName} ${User.lastName}`
                };
                return (<MenuTableRow
                  key={menuId}
                  item={item}
                />);
              })
            }
          </tbody>
        </table>
      </div>
    );
  }
}

MenuTable.propTypes = {
  menus: PropTypes.array.isRequired
};

export default MenuTable;
