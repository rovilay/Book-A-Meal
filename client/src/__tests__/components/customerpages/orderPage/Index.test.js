import React from 'react';
import { shallow } from 'enzyme';
import { CustomerOrderPage, mapStateToProps } from '../../../../components/customerpages/orderPage/Index';
import { userDataDefaultState } from '../../../../reducers/loginReducer';
import { filterDefaultState } from '../../../../reducers/filterReducer';
import { ordersDefaultState } from '../../../../reducers/ordersReducer';
import { modalDefaultState } from '../../../../reducers/modalReducer';
import { orders } from '../../../__mockData__/orderMock';


const props = {
  pagination: {
    limit: 12,
    count: 13,
    numOfPages: 2
  },
  setNav: jest.fn(),
  orders,
  history: {
    push: jest.fn()
  },
  modal: {},
  setModal: jest.fn(),
  updateOrder: jest.fn(),
  deleteMealInEditOrder: jest.fn(),
  updateOrderedMealPortion: jest.fn(),
  grandTotalPrice: 1100,
  getOrders: () => Promise.resolve(),
  getOrderMeals: () => Promise.resolve(),
  editOrder: jest.fn(),
  setEditOrder: jest.fn(),
  setFilter: jest.fn(),
  orderedMealsPagination:  {
    limit: 12,
    count: 13,
    numOfPages: 2
  }
};

describe('CustomerDashboard component test', () => {
  const setup = () => shallow(<CustomerOrderPage {...props} />);


  it('should render `CustomerOrderPage` correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should call `onEditOrder`', (done) => {
    const wrapper = setup();

    const onEditOrderSpy= jest.spyOn(wrapper.instance(), 'onEditOrder');
    wrapper.instance().onEditOrder(orders[0].Meals)
    expect(onEditOrderSpy).toHaveBeenCalled();

    done();
  });


  it('should call `showDetails`', (done) => {
    const wrapper = setup();

    const showDetailsSpy= jest.spyOn(wrapper.instance(), 'showDetails');
    wrapper.instance().showDetails(orders[0].Meals)
    expect(showDetailsSpy).toHaveBeenCalled();

    done();
  });


  it('should call `deleteRow`', (done) => {
    const wrapper = setup();


    const deleteRowSpy= jest.spyOn(wrapper.instance(), 'deleteRow');
    wrapper.instance().deleteRow('1')
    expect(deleteRowSpy).toHaveBeenCalled();

    done();
  });


  it('should call `updatePortion`', (done) => {
    const wrapper = setup();


    const updatePortionSpy= jest.spyOn(wrapper.instance(), 'updatePortion');
    wrapper.instance().updatePortion('1', 2)
    expect(updatePortionSpy).toHaveBeenCalled();

    done();
  });


  it('should call `getCustomerOrders`', (done) => {
    const wrapper = setup();
    localStorage.setItem('jwt', customerToken);

    const getCustomerOrdersSpy= jest.spyOn(wrapper.instance(), 'getCustomerOrders');
    wrapper.instance().getCustomerOrders(2)
    expect(getCustomerOrdersSpy).toHaveBeenCalled();

    done();
  });


  it('should call `hideModal`', (done) => {
    const wrapper = setup();

    const hideModalSpy= jest.spyOn(wrapper.instance(), 'hideModal');
    wrapper.instance().hideModal()
    expect(hideModalSpy).toHaveBeenCalled();

    done();
  });

  it('should call `handlePaginationClick`', (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePaginationClickSpy= jest.spyOn(wrapper.instance(), 'handlePaginationClick');
    wrapper.instance().handlePaginationClick(data)
    expect(handlePaginationClickSpy).toHaveBeenCalled();

    done();
  });

  it('should map state to props', () => {
    const initialState = {
      orders: {
        ...ordersDefaultState
      },
      modal: {
        ...modalDefaultState
      },
      login: {
        ...userDataDefaultState
      },
      filter: {
        ...filterDefaultState
      }
    };

    const ownProps = {
        pagination: {
          limit: 12,
          count: 13,
          numOfPages: 2
        },
        orders,
        modal: () => [],
        grandTotalPrice: 1100,
        orderedMealsPagination:  {
          limit: 12,
          count: 13,
          numOfPages: 2
        }
    };

    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  })
});
