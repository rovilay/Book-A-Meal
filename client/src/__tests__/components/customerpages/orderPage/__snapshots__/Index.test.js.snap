// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDashboard component test should map state to props 1`] = `
Object {
  "editOrder": Object {},
  "grandTotalPrice": 0,
  "modal": Object {
    "close": true,
    "content": Object {},
    "contentLabel": "",
    "isEdit": false,
    "isInfo": false,
    "isOpen": false,
    "isSetMenu": false,
    "pagination": Object {
      "limit": 10,
      "numOfPages": 1,
      "offset": 0,
    },
  },
  "orderedMealsPagination": Object {
    "count": 0,
    "limit": 5,
    "numOfPages": 1,
    "offset": 0,
  },
  "orders": Array [],
  "pagination": Object {
    "count": 0,
    "limit": 10,
    "numOfPages": 1,
    "offset": 0,
  },
  "userId": "",
}
`;

exports[`CustomerDashboard component test should render \`CustomerOrderPage\` correctly 1`] = `
<section
  className="cartpage orderpage"
>
  <div
    className="title merienda"
    id=""
  >
    Your Order History
  </div>
  <hr />
  <div
    className="table-container"
  >
    <FilterComp
      deleteMealInEditOrder={[MockFunction]}
      editOrder={[MockFunction]}
      getOrderMeals={[Function]}
      getOrders={[Function]}
      grandTotalPrice={1100}
      history={
        Object {
          "push": [MockFunction] {
            "calls": Array [
              Array [
                "/login",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        }
      }
      modal={Object {}}
      orderedMealsPagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setEditOrder={[MockFunction]}
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Menu",
                },
                Object {
                  "link": "/cart",
                  "title": "Cart",
                },
                Object {
                  "link": "/orders",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      tableContent="customer_orders"
      updateOrder={[MockFunction]}
      updateOrderedMealPortion={[MockFunction]}
    />
    <PaginationBoxView
      activeClassName="active"
      breakClassName="break-me"
      breakLabel={
        <a
          href=""
        >
          ...
        </a>
      }
      containerClassName="pagination"
      disableInitialCallback={false}
      disabledClassName="disabled"
      marginPagesDisplayed={2}
      nextClassName="next"
      nextLabel=">>"
      onPageChange={[Function]}
      pageCount={2}
      pageRangeDisplayed={5}
      previousClassName="previous"
      previousLabel="<<"
      subContainerClassName="pages pagination"
    />
  </div>
  <div
    className="container-test"
  >
    <div
      className="row head"
    >
      <p
        className="row-item"
        key="0"
      >
        S/N
      </p>
      <p
        className="row-item"
        key="1"
      >
        OrderId
      </p>
      <p
        className="row-item"
        key="2"
      >
        Date
      </p>
      <p
        className="row-item"
        key="3"
      >
        Total Cost
      </p>
      <p
        className="row-item"
        key="4"
      >
        Actions
      </p>
    </div>
    <CustomerOrderTableRow
      actions={
        Object {
          "delete": true,
          "edit": true,
          "info": true,
        }
      }
      deleteMealInEditOrder={[MockFunction]}
      editOrder={[Function]}
      getOrderMeals={[Function]}
      getOrders={[Function]}
      grandTotalPrice={1100}
      history={
        Object {
          "push": [MockFunction] {
            "calls": Array [
              Array [
                "/login",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        }
      }
      isEdit={false}
      item={
        Object {
          "date": "August 23, 2018",
          "orderId": 1,
          "sn": NaN,
          "totalPrice": undefined,
        }
      }
      key="1"
      mealsUrl="api/v1/orders/1"
      modal={Object {}}
      orderCreatedAt="2018-08-23"
      orderedMealsLength={1}
      orderedMealsPagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setEditOrder={[MockFunction]}
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Menu",
                },
                Object {
                  "link": "/cart",
                  "title": "Cart",
                },
                Object {
                  "link": "/orders",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      showDetails={[Function]}
      sn={2}
      updateOrder={[MockFunction]}
      updateOrderedMealPortion={[MockFunction]}
      updatePortion={[Function]}
    />
    <CustomerOrderTableRow
      actions={
        Object {
          "delete": true,
          "edit": true,
          "info": true,
        }
      }
      deleteMealInEditOrder={[MockFunction]}
      editOrder={[Function]}
      getOrderMeals={[Function]}
      getOrders={[Function]}
      grandTotalPrice={1100}
      history={
        Object {
          "push": [MockFunction] {
            "calls": Array [
              Array [
                "/login",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        }
      }
      isEdit={false}
      item={
        Object {
          "date": "August 24, 2018",
          "orderId": 2,
          "sn": NaN,
          "totalPrice": undefined,
        }
      }
      key="2"
      mealsUrl="api/v1/orders/2"
      modal={Object {}}
      orderCreatedAt="2018-08-24"
      orderedMealsLength={1}
      orderedMealsPagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setEditOrder={[MockFunction]}
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Menu",
                },
                Object {
                  "link": "/cart",
                  "title": "Cart",
                },
                Object {
                  "link": "/orders",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      showDetails={[Function]}
      sn={3}
      updateOrder={[MockFunction]}
      updateOrderedMealPortion={[MockFunction]}
      updatePortion={[Function]}
    />
    <CustomerOrderTableRow
      actions={
        Object {
          "delete": true,
          "edit": true,
          "info": true,
        }
      }
      deleteMealInEditOrder={[MockFunction]}
      editOrder={[Function]}
      getOrderMeals={[Function]}
      getOrders={[Function]}
      grandTotalPrice={1100}
      history={
        Object {
          "push": [MockFunction] {
            "calls": Array [
              Array [
                "/login",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        }
      }
      isEdit={false}
      item={
        Object {
          "date": "August 25, 2018",
          "orderId": 3,
          "sn": NaN,
          "totalPrice": undefined,
        }
      }
      key="3"
      mealsUrl="api/v1/orders/3"
      modal={Object {}}
      orderCreatedAt="2018-08-25"
      orderedMealsLength={1}
      orderedMealsPagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setEditOrder={[MockFunction]}
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Menu",
                },
                Object {
                  "link": "/cart",
                  "title": "Cart",
                },
                Object {
                  "link": "/orders",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      showDetails={[Function]}
      sn={4}
      updateOrder={[MockFunction]}
      updateOrderedMealPortion={[MockFunction]}
      updatePortion={[Function]}
    />
  </div>
  <div
    className="order"
  >
    <p
      className="grandTot"
    >
      Grand Total (₦): 
      1100
    </p>
  </div>
  <ModalComp
    deleteMealInEditOrder={[MockFunction]}
    deleteRow={[Function]}
    editOrder={[MockFunction]}
    getOrderMeals={[Function]}
    getOrders={[Function]}
    grandTotalPrice={1100}
    hideModal={[Function]}
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/login",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    modal={Object {}}
    orderedMealsPagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    orders={
      Array [
        Object {
          "Meals": "api/v1/orders/1",
          "createdAt": "2018-08-23",
          "id": 1,
        },
        Object {
          "Meals": "api/v1/orders/2",
          "createdAt": "2018-08-24",
          "id": 2,
        },
        Object {
          "Meals": "api/v1/orders/3",
          "createdAt": "2018-08-25",
          "id": 3,
        },
      ]
    }
    pagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    setEditOrder={[MockFunction]}
    setFilter={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "filter": "all",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setModal={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "close": true,
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setNav={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Menu",
              },
              Object {
                "link": "/cart",
                "title": "Cart",
              },
              Object {
                "link": "/orders",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    updateOrder={[MockFunction]}
    updateOrderedMealPortion={[MockFunction]}
    updatePortion={[Function]}
  />
</section>
`;
